cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1433,11 +1455,12 @@ bool IsDirectory(std::string_view key, const S3Model::HeadObjectResult& result)
 class ObjectInputFile final : public io::RandomAccessFile {
  public:
   ObjectInputFile(std::shared_ptr<S3ClientHolder> holder, const io::IOContext& io_context,
-                  const S3Path& path, int64_t size = kNoSize)
+                  const S3Path& path, int64_t size = kNoSize, const std::string& c_key = "")
```

Comment:
Could you use `sse_customer_key` not `c_key`?

Reply:
ok, I would refine.